libfwupdplugin/fu-device.c

```diff
@@ -4731,10 +4751,18 @@ fu_device_init (FuDevice *self)
 	priv->parent_guids = g_ptr_array_new_with_free_func (g_free);
 	priv->possible_plugins = g_ptr_array_new_with_free_func (g_free);
 	priv->retry_recs = g_ptr_array_new_with_free_func (g_free);
+	priv->progress = fu_progress_new();
 	g_rw_lock_init (&priv->parent_guids_mutex);
 	g_rw_lock_init (&priv->metadata_mutex);
 	priv->notify_flags_handler_id =
 	    g_signal_connect(self, "notify::flags", G_CALLBACK(fu_device_flags_notify_cb), NULL);
+
+	/* allows more verbose profiling for plugin development */
+	fu_progress_set_profile(priv->progress, g_getenv("FWUPD_PROGRESS_PROFILE") != NULL);
```

Comment:
Some suggestions to consider here...
1. Maybe make `fwupdtool` always set this.  People do development using fwupdtool, and get this data.
2. If a plugin doesn't set custom steps, then assume that it needs to be profiled.

Reply:
Yes, I agree that we need to set the profiling on more aggressively. Maybe we also tie it to the daemon --verbose option?

> If a plugin doesn't set custom steps, then assume that it needs to be profiled.

Maybe. That needs some thinking about because the non-custom step count is typically set before a loop of some sort. Will ponder.